include/net/net_if.h

```diff
@@ -782,6 +782,10 @@ static inline int net_if_set_link_addr(struct net_if *iface,
  */
 static inline uint16_t net_if_get_mtu(struct net_if *iface)
 {
+	if (iface == NULL) {
+		return 0U;
+	}
+
 	return iface->if_dev->mtu;
 }
```

Comment:
> I don't think that this fixes #34000. It might silence Coverity warning for some reason, but I don't see direct connection between this change and reported Coverity warning.

It fixes the issue because the mtu function now checks the NULL and does not crash.

The NULL check in `net_if_flag_is_set()` is a valid issue as I saw a crash when testing these. Adding the checks to MTU functions are useful to have as these could be called by the application and we are not calling these functions constantly so there is no performance concerns. IMHO we should add more NULL checks in various network functions as the asserts are not enough.

Reply:
Sorry for not getting back to this PR for so long.

>The NULL check in `net_if_flag_is_set()` is a valid issue as I saw a crash when testing these.

That is one point of view. I see it in a bit different way, i.e. any crash is a result of bug, which is not checking for `iface != NULL`. Such crash is quite easy to debug and fix, compared to continuing execution with buggy logic.

>IMHO we should add more NULL checks in various network functions as the asserts are not enough.

Why are asserts not enough? Because they are not compiled in by default?

Simply checking for NULL parameters and returning arbitrary values in case of NULL doesn't solve all issues. It solves crashes in net_if code (like in `net_if_flag_is_set()`), but it doesn't make higher level code that executes net_if functions any better. Functions like `net_if_flag_is_set()` are in most cases called with other net_if functions. Should we check `iface` argument in all net_if functions? Additionally `net_if_flag_is_set(NULL, flag) == 0` logic doesn't have to be always correct. Take into account this fragment:

https://github.com/zephyrproject-rtos/zephyr/blob/facd0f57db85163432bdca7767a05e3f1d742b9f/subsys/net/ip/net_if.c#L460-L467

If someone will use `if (!net_if_flag_is_set(iface, FLAG))` (notice negation `!`), then we need to make sure to not dereference `iface` later on, as it might still be `NULL`. I think that it is easier to make sure `iface != NULL` from the beginning and leave `net_if_flag_is_set()` API as is, without adding NULL check resulting in (slightly, but still) increased flash size and (small) performance penalty. IMO crashes because of NULL pointer dereferencing are often easier to debug than some issues in logic, which could happen after treating NULL in functions like `net_if_flag_is_set()` as a "valid" interface.